casper/src/main/scala/coop/rchain/casper/storage/LmdbKeyValueStore.scala

```diff
@@ -49,39 +49,68 @@ final case class LmdbKeyValueStore[F[_]: Sync](
                }
     } yield result
 
-  def withReadTxnF[T](f: (Txn[ByteBuffer], Dbi[ByteBuffer]) => F[T]): F[T] =
-    withTxn(isWrite = false)(f)
+  // Ensures transaction is used only on one thread.
```

Comment:
This fix is included on purpose?

Reply:
Yes, I forgot to mention. I included this fix for tests to pass. Before merging I'll rebase it.